go-controller/pkg/ovn/ovn.go

```diff
@@ -225,84 +258,187 @@ func getPodNamespacedName(pod *kapi.Pod) string {
-		retryNetworkPolicies:      NewRetryObjs(factory.PolicyType, "", nil, nil, nil),
-		retryNodes:                NewRetryObjs(factory.NodeType, "", nil, nil, nil),
-		retryEgressFirewalls:      NewRetryObjs(factory.EgressFirewallType, "", nil, nil, nil),
-		retryEgressIPs:            NewRetryObjs(factory.EgressIPType, "", nil, nil, nil),
-		retryEgressIPNamespaces:   NewRetryObjs(factory.EgressIPNamespaceType, "", nil, nil, nil),
-		retryEgressIPPods:         NewRetryObjs(factory.EgressIPPodType, "", nil, nil, nil),
-		retryEgressNodes:          NewRetryObjs(factory.EgressNodeType, "", nil, nil, nil),
-		retryCloudPrivateIPConfig: NewRetryObjs(factory.CloudPrivateIPConfigType, "", nil, nil, nil),
-		retryNamespaces:           NewRetryObjs(factory.NamespaceType, "", nil, nil, nil),
-		recorder:                  recorder,
-		nbClient:                  libovsdbOvnNBClient,
-		sbClient:                  libovsdbOvnSBClient,
-		svcController:             svcController,
-		svcFactory:                svcFactory,
-		egressSvcController:       egressSvcController,
-		podRecorder:               metrics.NewPodRecorder(),
+		}
+		oc.externalGWCache = make(map[ktypes.NamespacedName]*externalRouteInfo)
+		oc.exGWCacheMutex = sync.RWMutex{}
+
+		oc.retryEgressFirewalls = NewRetryObjs(factory.EgressFirewallType, "", nil, nil, nil)
+		oc.retryEgressIPs = NewRetryObjs(factory.EgressIPType, "", nil, nil, nil)
+		oc.retryEgressIPNamespaces = NewRetryObjs(factory.EgressIPNamespaceType, "", nil, nil, nil)
+		oc.retryEgressIPPods = NewRetryObjs(factory.EgressIPPodType, "", nil, nil, nil)
+		oc.retryEgressNodes = NewRetryObjs(factory.EgressNodeType, "", nil, nil, nil)
+		oc.retryCloudPrivateIPConfig = NewRetryObjs(factory.CloudPrivateIPConfigType, "", nil, nil, nil)
+
+		if addressSetFactory == nil {
+			addressSetFactory = addressset.NewOvnAddressSetFactory(cm.nbClient)
+		}
+		oc.addressSetFactory = addressSetFactory
+		oc.sharedNetpolPortGroups = syncmap.NewSyncMap[*defaultDenyPortGroups]()
+		oc.namespaces = make(map[string]*namespaceInfo)
+		oc.namespacesMutex = sync.Mutex{}
+		oc.retryNamespaces = NewRetryObjs(factory.NamespaceType, "", nil, nil, nil)
+		oc.retryNetworkPolicies = NewRetryObjs(factory.PolicyType, "", nil, nil, nil)
+
+		oc.wg = cm.defaultWg
+		oc.multicastSupport = config.EnableMulticast
+		cm.defaultController = oc
+	} else {
+		oc.wg = &sync.WaitGroup{}
+		oc.multicastSupport = false // disable multicast support for now
+	}
+	_, loaded := cm.allOvnControllers.LoadOrStore(nadInfo.NetName, oc)
+	if loaded {
+		klog.Errorf("Network attachment definition %s already exists", nadInfo.NetName)
 	}
+	return oc, nil
 }
 
-// Run starts the actual watching.
-func (oc *Controller) Run(ctx context.Context, wg *sync.WaitGroup) error {
-	// Start and sync the watch factory to begin listening for events
-	if err := oc.watchFactory.Start(); err != nil {
+func (oc *Controller) Init(ctx context.Context) error {
```

Comment:
This method should be called Start or something that connotes the fact that this controller starts running here. This is not an initialization.